File: SuperiorPackGroup/Code Classes/BomDemand.vb
Imports DXDAL.SPGData

Public Class BomDemand

    Public Property ItemID As Items
    Public Property ItemType As String
    Public Property PoolItemID As ItemPool
    Public Property ProjectDetailID As ProjectDetails
    Public Property ProjectID As Project
    Public property QuantityOnHand As Double
    Public property QuantityOnScheduledProjects As Double
    Public Property QuantityOnUnscheduledProjects As Double
    Public Property QuantityRequiredForProject As Double

    Public ReadOnly Property BomDemandShortage As Double
        Get
            Return Math.Max(QuantityRequiredForProject - (QuantityOnHand - QuantityOnScheduledProjects), 0)
        End Get
    End Property

    Public ReadOnly Property CustomerPO As String
        Get
            If ProjectDetailID Is Nothing Then
                Return String.Empty
            Else
                Return ProjectDetailID.CustomerPO
            End If
        End Get
    End Property

    Public ReadOnly Property ItemCode As String
        Get
            If ItemID IsNot Nothing Then
                Return ItemID.ItemCode
            End If
            If PoolItemID IsNot Nothing Then
                Return PoolItemID.PoolCode
            End If
            Return String.Empty
        End Get
    End Property

    Public ReadOnly Property ItemDescription As String
        Get
            If ItemID IsNot Nothing Then
                Return ItemID.ItemDescription
            End If
            If PoolItemID IsNot Nothing Then
                Return PoolItemID.PoolDescription
            End If
            Return String.Empty
        End Get
    End Property

End Class

File: SuperiorPackGroup/Code Classes/BomDemand.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using DXDAL.SPGData;

namespace SuperiorPackGroup
{
	public class BomDemand
	{

		public Items ItemID {get; set;}
		public string ItemType {get; set;}
		public ItemPool PoolItemID {get; set;}
		public ProjectDetails ProjectDetailID {get; set;}
		public Project ProjectID {get; set;}
		public double QuantityOnHand {get; set;}
		public double QuantityOnScheduledProjects {get; set;}
		public double QuantityOnUnscheduledProjects {get; set;}
		public double QuantityRequiredForProject {get; set;}

		public double BomDemandShortage
		{
			get
			{
				return Math.Max(QuantityRequiredForProject - (QuantityOnHand - QuantityOnScheduledProjects), 0);
			}
		}

		public string CustomerPO
		{
			get
			{
				if (ProjectDetailID == null)
				{
					return string.Empty;
				}
				else
				{
					return ProjectDetailID.CustomerPO;
				}
			}
		}

		public string ItemCode
		{
			get
			{
				if (ItemID != null)
				{
					return ItemID.ItemCode;
				}
				if (PoolItemID != null)
				{
					return PoolItemID.PoolCode;
				}
				return string.Empty;
			}
		}

		public string ItemDescription
		{
			get
			{
				if (ItemID != null)
				{
					return ItemID.ItemDescription;
				}
				if (PoolItemID != null)
				{
					return PoolItemID.PoolDescription;
				}
				return string.Empty;
			}
		}

	}

}